packages/web3-providers-http/src/index.ts

```diff
@@ -0,0 +1,58 @@
+import axios, { AxiosInstance } from 'axios';
+import Web3ProviderBase from 'web3-providers-base';
+import { ProviderOptions } from 'web3-providers-base/types';
+
+import { HttpRpcOptions, HttpRpcResponse, IWeb3Provider } from '../types';
+
+export default class Web3ProvidersHttp extends Web3ProviderBase implements IWeb3Provider {
+  private _httpClient: AxiosInstance | undefined
+
+  constructor(options: ProviderOptions) {
+    super(options);
+    this._httpClient = Web3ProvidersHttp.createHttpClient(options.providerString);
+  }
+
+  static validateProviderString(providerString: string): boolean {
+    try {
+      return (typeof providerString !== 'string'
+          || /^http(s)?:\/\//i.test(providerString));
+    } catch (error) {
+      throw Error(`Failed to validate provider string: ${error.message}`);
+    }
+  }
+
+  static createHttpClient(baseUrl: string): AxiosInstance {
```

Comment:
Would users want to create an httpclient outside of this scope of the class itself?

Reply:
I don't see why, but I did this so the same method would be copied in every instance of the class, but that probably doesn't matter?

Reply:
In my mind its only ever called in the constructor? I'm trying to think of other use cases

Reply:
So by making `validateProviderUrl` and `createHttpClient` `private`, we loose the ability to test those methods. I'm going to revert these changes and keep the use of `static` as these methods don't belong to instances, but rather the class itself